### PR TITLE
Add combocid dispatch from QD to QE

### PR DESCRIPTION
**Pull request by lss602726449**
In order to ensure visibility within transactions on QE, we dispatch combocid from QD to QE. We add combocid_hook to dispatch combocid in the CommandCounterIncrement, when an SQL ends. We dispatch combocid in an incremental manner. In order to ensure that combocid can be correctly recognized by QE, we record combocid and HEAP_COMBOCID flag in WAL, and restore them when replaying the log, especially for delete and update.

closes: #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to @cloudberrydb/dev team for review and approval when your PR is ready🥳

### CODE REVIEW

File: src/backend/access/transam/xact.c [hunk 1]
```diff
@@ -95,6 +95,9 @@
 #include "utils/vmem_tracker.h"
 #include "cdb/cdbdisp.h"
 #include "postmaster/autovacuum.h"
+#include "utils/combocid.h"
+
+combocid_hook_type combocid_hook = NULL;
```
Comment (my-ship-it) [inline src/backend/access/transam/xact.c:100]: We need to change the name from combocid to CCI_hook?

File: src/backend/access/heap/heapam.c [hunk 2]
```diff
@@ -9048,7 +9052,7 @@ heap_xlog_delete(XLogReaderState *record)
 		else
 			HeapTupleHeaderSetXmin(htup, InvalidTransactionId);
 
-		HeapTupleHeaderSetCmax(htup, xlrec->t_cid, false);
+		HeapTupleHeaderSetCid(htup, xlrec->t_cid);
```
Comment (HuSen8891) [inline src/backend/access/heap/heapam.c:9055]: HeapTupleHeaderSetCid(htup, xlrec->t_cid) --> HeapTupleHeaderSetCmax(htup, xlrec->t_cid, htup->t_infomask & HEAP_COMBOCID);

File: src/backend/access/heap/heapam.c [hunk 3]
```diff
@@ -9566,7 +9570,7 @@ heap_xlog_update(XLogReaderState *record, bool hot_update)
 		htup->t_hoff = xlhdr.t_hoff;
 
 		HeapTupleHeaderSetXmin(htup, XLogRecGetXid(record));
-		HeapTupleHeaderSetCmin(htup, xlhdr.t_cid);
+		HeapTupleHeaderSetCid(htup, xlhdr.t_cid);
```
Comment (HuSen8891) [inline src/backend/access/heap/heapam.c:9573]: this should not change, cmin should never be a combo id.

File: src/backend/access/heap/heapam.c [hunk 4]
```diff
@@ -9432,7 +9436,7 @@ heap_xlog_update(XLogReaderState *record, bool hot_update)
 		fix_infomask_from_infobits(xlrec->old_infobits_set, &htup->t_infomask,
 								   &htup->t_infomask2);
 		HeapTupleHeaderSetXmax(htup, xlrec->old_xmax);
-		HeapTupleHeaderSetCmax(htup, xlrec->t_cid, false);
+		HeapTupleHeaderSetCid(htup, xlrec->t_cid);
```
Comment (HuSen8891) [inline src/backend/access/heap/heapam.c:9439]: HeapTupleHeaderSetCid(htup, xlrec->t_cid) --> HeapTupleHeaderSetCmax(htup, xlrec->t_cid, htup->t_infomask & HEAP_COMBOCID);

File: src/backend/access/heap/heapam.c [hunk 5]
```diff
@@ -9292,7 +9296,7 @@ heap_xlog_multi_insert(XLogReaderState *record)
 			htup->t_infomask = xlhdr->t_infomask;
 			htup->t_hoff = xlhdr->t_hoff;
 			HeapTupleHeaderSetXmin(htup, XLogRecGetXid(record));
-			HeapTupleHeaderSetCmin(htup, xlhdr->t_cid);
+			HeapTupleHeaderSetCid(htup, xlhdr->t_cid);
```
Comment (HuSen8891) [inline src/backend/access/heap/heapam.c:9299]: this should not change, cmin should never be a combo id.

File: src/backend/access/transam/xact.c (hunk 1 above)
Comment (lss602726449) [reply src/backend/access/transam/xact.c:100]: OK, I'll change the hook name later

File: src/backend/access/heap/heapam.c (hunk 5 above)
Comment (lss602726449) [reply src/backend/access/heap/heapam.c:9299]: Thanks for your serveral suggestion. I'll change the code later

**Pull request closed without merging**